### PR TITLE
244 - Integration download dataset

### LINKED ISSUE #244: Download all dataset files
## Overview of the Feature Request
For downloading all dataset files, we can use the existing /access/dataset/{id}/versions/{versionId}endpoint: [API Guide](https://guides.dataverse.org/en/latest/api/dataaccess.html#download-by-dataset-by-version). It is very similar to the previous one described, and also provides the option to download original or archival files.

This is the frontend component to integrate with the API:
- https://github.com/IQSS/dataverse-frontend/issues/188

## What kind of user is the feature intended for?
All of them

## What inspired the request?
- https://github.com/IQSS/dataverse/issues/9997

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
Be able to download the all dataset files (archival or original formats) using the Download button in the dataset page.

## Any open or closed issues related to this feature request?
- https://github.com/IQSS/dataverse/issues/9997

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR adds the integration to download the dataset using the Dataverse API. With this integration the user would be able to download the dataset both in original and archival formats using the Access Dataset button in the Dataset page.

## Which issue(s) this PR closes:

- Closes #244 

### Step 1: Run the development environment
1. Run `npm I`
3. `cd packages/design-system && npm run build`
4. `cd ../../`
5. Check that you have a  `.env` file such as the .env.example, with the `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000` variable
6.  `cd dev-env`
7. `./run-env.sh unstable`
8. To check the environment go to <http://localhost:8000> and check your local dataverse installation

### Step 2: Test Dataset View mode with the implemented changes to download the files
1. Go to <http://localhost:8000>
2. Login as admin using username: dataverseAdmin and password: admin1
3. Create a new Dataset
4. Upload some files to the dataset
5. From the dataset view mode copy the search parameters from the url. Ex.: `?persistentId=doi:10.5072/FK2/LHGRHP&version=DRAFT`
6. Go to <http://localhost:8000/spa/datasets> and paste the search parameters. Ex.: <http://localhost:8000/spa/datasets?persistentId=doi:10.5072/FK2/LHGRHP&version=DRAFT>
8. You are currently viewing the same dataset in the SPA. In the files table, locate the "Access Dataset" dropdown on the right of the page. Use the button to download the dataset and check that is working as expected.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No

## Is there a release notes update needed for this change?:
No

### CODE REVIEW

**Comment by GPortas** [comment]
@MellyGray Can you please resolve conflicts?

**Comment by MellyGray** [comment]
@GPortas conflicts resolved

File: src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts [hunk 1]
```diff
@@ -76,10 +95,12 @@ export class DatasetJSDataverseRepository implements DatasetRepository {
             canManageDatasetPermissions: true,
             canDeleteDatasetDraft: true,
             canViewUnpublishedDataset: true
-          },
-          []
+          }, // TODO Connect with JS dataset permissions for privateUrl when it is available in js-dataverse
+          [], // TODO Connect with JS dataset locks for privateUrl when it is available in js-dataverse
+          0, // TODO Connect with JS dataset filesTotalDownloadSize for privateUrl when it is available in js-dataverse
+          0 // TODO Connect with JS dataset filesTotalDownloadSize for privateUrl when it is available in js-dataverse
```
Comment (GPortas) [inline src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts:101]: Can we create an issue including all these TODOs related to private URLs?

**Review by GPortas — COMMENTED** [review]
Looks good!

https://github.com/IQSS/dataverse-frontend/assets/17010447/5b2f9349-a56d-4bdf-94b3-d8dffc69203c

**Comment by GPortas** [comment]
@MellyGray Can you please resolve merge conflicts?

File: src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts (hunk 1 above)
Comment (MellyGray) [reply src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts:101]: I created an issue
- https://github.com/IQSS/dataverse-frontend/issues/266

**Comment by MellyGray** [comment]
@GPortas conflicts resolved

**Review by GPortas — APPROVED** [review]

**Pull request merged**